carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -259,6 +264,50 @@ void WMBroadcaster::addPassingControlLineFromMsg(std::shared_ptr<Geofence> gf_pt
   gf_ptr->regulatory_element_ = std::make_shared<lanelet::PassingControlLine>(lanelet::PassingControlLine::buildData(
     lanelet::utils::getId(), pcl_bounds, left_participants, right_participants));
 }
+
+void WMBroadcaster::addRegionAccessRule(std::shared_ptr<Geofence> gf_ptr, const cav_msgs::TrafficControlMessageV01& msg_v01, const std::vector<lanelet::Lanelet>& affected_llts) const
+{
+  ros::V_string participants;
+ for (j2735_msgs::TrafficControlVehClass participant : msg_v01.params.vclasses)
+  {
+    // Currently j2735_msgs::TrafficControlVehClass::RAIL is not supported
+    if (participant.vehicle_class == j2735_msgs::TrafficControlVehClass::ANY)
+    {
+      participants = {lanelet::Participants::Vehicle, lanelet::Participants::Pedestrian, lanelet::Participants::Bicycle};
+      break;
+    }
+    else if (participant.vehicle_class == j2735_msgs::TrafficControlVehClass::PEDESTRIAN)
+    {
+      participants.push_back(lanelet::Participants::Pedestrian);
+    }
+    else if (participant.vehicle_class == j2735_msgs::TrafficControlVehClass::BICYCLE)
+    {
+      participants.push_back(lanelet::Participants::Bicycle);
+    }
+    else if (participant.vehicle_class == j2735_msgs::TrafficControlVehClass::MICROMOBILE ||
+              participant.vehicle_class == j2735_msgs::TrafficControlVehClass::MOTORCYCLE)
+    {
+      participants.push_back(lanelet::Participants::VehicleMotorcycle);
+    }
+    else if (participant.vehicle_class == j2735_msgs::TrafficControlVehClass::BUS)
+    {
+      participants.push_back(lanelet::Participants::VehicleBus);
+    }
+    else if (participant.vehicle_class == j2735_msgs::TrafficControlVehClass::LIGHT_TRUCK_VAN ||
+            participant.vehicle_class == j2735_msgs::TrafficControlVehClass::PASSENGER_CAR)
+    {
+      participants.push_back(lanelet::Participants::VehicleCar);
+    }
+    else if (8<= participant.vehicle_class && participant.vehicle_class <= 16) // Truck enum definition range from 8-16 currently
+    {
+      participants.push_back(lanelet::Participants::VehicleTruck);
+    }
+  }
```

Comment:
we also have this big logic up in addPassingControlLineFromMsg
So let's factor this out